frontend/subrequests/lint/lint.go

```diff
@@ -124,47 +110,69 @@ func (results *LintResults) ToResult() (*client.Result, error) {
 	return res, nil
 }
 
+func (results *LintResults) validateWarnings() error {
+	for _, warning := range results.Warnings {
+		if warning.Location.SourceIndex < 0 {
+			return fmt.Errorf("sourceIndex is required")
```

Comment:
Missing source should not cause the program to error but skip printing the source mapping(-1 can mean no mapping). With missing ranges we can print the File properties but not the source lines(for the actual `Print()` function this should already be handled there, but not for sorting.

Missing Definition is just ok, as it is not used by printing.

Reply:
👍